### PR TITLE
templates - sync from community-templates

### PR DESCRIPTION
**Pull request by tbrisker**
@ares @lzap @mmoll - I would appreciate your help ensuring that all the snapshot changes are expected

### CODE REVIEW

**Comment by tbrisker** [comment]
Fixed some of the tests, but looks like the snapshot test is failing for some reason - maybe because the pxegrub2_chainload snippet is loaded and tested before the pxegrub2_discovery snippet?

File: app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb [hunk 1]
```diff
@@ -5,17 +5,26 @@ name: iPXE intermediate script
...
 #!ipxe
-# Intermediate iPXE script to report MAC address to Foreman
 
-<% (0..32).each do |i| -%>
-:net<%= i %>
-isset ${net<%= i -%>/mac} || goto no_nic
-dhcp net<%= i -%> || goto net<%= i + 1 %>
-chain <%= foreman_url('iPXE') %>?mac=${net<%= i -%>/mac} || goto net<%= i + 1 %>
+# Skips booting from network and continues booting from next device
+exit
```
Comment (timogoebel) [inline app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: I belive this change does not make sense. If I‘m not mistaken the change in the core repo is the one we want to have.

Comment (tbrisker) [reply app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: Could you please open a PR to community templates with it? Look like in core it was added by https://github.com/theforeman/foreman/pull/6061 while in community templates a different version was added in https://github.com/theforeman/community-templates/pull/512.

Comment (timogoebel) [reply app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: https://github.com/theforeman/community-templates/pull/578

Comment (tbrisker) [reply app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: Thanks, updated. Any idea about the snapshots failing on `pxegrub2_chainload`?

Comment (lzap) [reply app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: Taking a look, in the meantime I created a PR for the typo: https://github.com/theforeman/community-templates/pull/579

Comment (timogoebel) [reply app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: It tries to find the snipped with the source classs `Foreman::Renderer::Source::Database`, however it should use `SnapshotSource` instead.

You can debug that with:
```ruby
diff --git a/lib/foreman/renderer/scope/macros/snippet_rendering.rb b/lib/foreman/renderer/scope/macros/snippet_rendering.rb
index d5db6f787..adae111d0 100644
--- a/lib/foreman/renderer/scope/macros/snippet_rendering.rb
+++ b/lib/foreman/renderer/scope/macros/snippet_rendering.rb
@@ -15,6 +15,7 @@ module Foreman
           def snippet(name, options = {}, variables: {})
             template = source.find_snippet(name)
             unless template
+              binding.pry if name == 'pxegrub2_discovery'
               raise "The specified snippet '#{name}' does not exist, or is not a snippet." unless options[:silent]
               return
             end
```

I haven't figured out yet, where the wrong source comes from.

Comment (lzap) [reply app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: I am looking into another problem - discovery menus are nested by accident.

Comment (timogoebel) [reply app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: The `Foreman::Renderer::Source::Database` source comes from this:
https://github.com/theforeman/foreman/blob/3f4943103cf73274c827bfe61b7b14cc4ff0c2fb/lib/foreman/renderer.rb#L31-L33

However I ran out of time for now and wasn't able to provide a fix for this.

**Comment by lzap** [comment]
Fixed relevant issue with nesting: https://github.com/theforeman/community-templates/pull/580

Now I will try to help @timogoebel to debug the issue. Obviously discovery snippet now includes a snippet so there is now template-snippet-snippet chain which might be the issue.

File: app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb (hunk 1 above)
Comment (lzap) [reply app/views/unattended/provisioning_templates/iPXE/ipxe_intermediate_script.erb]: One observation: pxegrub2_discovery is now included from another snippet named pxegrub2_chainload. Can this be the issue (snippet-snippet context wrong)?

**Comment by lzap** [comment]
Guys, I am filing new bug here https://projects.theforeman.org/issues/26612 to track the issue and I am going to refactor templates to avoid snippet in snippet shortly (after a short meeting).

**Review by lzap — DISMISSED** [review]
Other than that looks good. There is a big change - wget/curl POST request, it has been tested but mostly on CentOS so there might be dragons. Hopefully we can catch the rest in the test week.

File: test/unit/foreman/renderer/snapshots/ProvisioningTemplate/iPXE/iPXE intermediate script.snap.txt [hunk 2]
```diff
@@ -5,167 +5,137 @@
 isset ${net0/mac} || goto no_nic
 dhcp net0 || goto net1
 chain http://foreman.some.host.fqdn/unattended/iPXE?mac=${net0/mac} || goto net1
-
```
Comment (timogoebel) [inline test/unit/foreman/renderer/snapshots/ProvisioningTemplate/iPXE/iPXE intermediate script.snap.txt]: I guess we want to keep the newline here, right?

**Comment by timogoebel** [comment]
Just one nit, otherwise looks good to me.

File: test/unit/foreman/renderer/snapshots/ProvisioningTemplate/iPXE/iPXE intermediate script.snap.txt (hunk 2 above)
Comment (tbrisker) [reply test/unit/foreman/renderer/snapshots/ProvisioningTemplate/iPXE/iPXE intermediate script.snap.txt]: could you fix the relevant template in community templates please? This is auto-generated from it :)

**Comment by lzap** [comment]
Looks good (the rebased version). Thanks.

File: test/unit/foreman/renderer/snapshots/ProvisioningTemplate/iPXE/iPXE intermediate script.snap.txt (hunk 2 above)
Comment (mmoll) [reply test/unit/foreman/renderer/snapshots/ProvisioningTemplate/iPXE/iPXE intermediate script.snap.txt]: https://github.com/theforeman/community-templates/pull/582

**Comment by tbrisker** [comment]
Thanks @lzap and @timogoebel ! tests have now passed, merging.

**Pull request merged**